### DISCUSSION TITLE
cache docs

### DISCUSSION THREAD

**Discussion by nilsnolde** [valhalla/valhalla#3320] [Q&A] [answered]
while working on #3281 I realized I have little idea about the caching mechanisms in valhalla or even more, what effects it will have to use one cache over the other.

I understand the mechanics of flat, simple and (sorta) LRU cache, but wondering:
- what's the difference between flat & simple? simple is just a bit quicker accessing tiles with its unordered map but a little more ram intense?
- what's the use case for `SynchronizedTileCache`? is that for kinda ram-restricted devices where you still want to multi-thread but without setting up a separate cache for each thread at the expense of slightly slower cache access bcs of the locking?

if I have a better understanding I can contribute a brief docs section in docs/baldr.md. been wondering about this for a longer time

**Comment by kevinkreiser** [comment, marked as answer]
sorry for the wait on this one.

basically we have both shared and unshared caches when it comes how threads and even processes can share tile caches. for processes the only way for them to share tile cache is through a memory map of a tar file. for threads they can share tile cache either through the memory mapped tar OR through the synchronized tile cache which uses individual files loaded into memory from the tile_dir. if you dont use the synchronized tile cache and you are using the tile_dir then threads or processes will indeed all have their own caches, essentially duplicating them for each.
